crates/cranelift/src/compiler.rs

```diff
@@ -31,6 +31,12 @@ use wasmtime_environ::{
     TrapEncodingBuilder, TrapInformation, Tunables, TypeTables, VMOffsets,
 };
 
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
+pub(crate) const CODE_SECTION_ALIGNMENT: u64 = 0x4000;
+
+#[cfg(not(all(target_os = "macos", target_arch = "aarch64")))]
+pub(crate) const CODE_SECTION_ALIGNMENT: u64 = 0x1000;
```

Comment:
At least on macOS you have to query the page size at runtime. This is for example necessary for Rosetta 2 to work as it uses AArch64 sized pages. There may be other cases where it isn't possible to know this at compile time. You can use `region::page::size()` to query it.

Reply:
TIL! @alexcrichton in this case, would it be just simpler to call `region::page::size()` (maybe cache it, if we expect this to be expensive) instead of having the raw constants scattered in the code?

Reply:
I think a better fix for this would be to inspect `self.isa.target()` to accomodate cross-compiling use cases. If the section is over-aligned for some embeddings that's not an issue.